plugins/schedule/base/services/ResourceUserService.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * The ResourceUser service enables you to create and manage associations between users and resources.
+ * @service resourceUser
+ * @package plugins.schedule
+ * @subpackage api.services
+ */
+class ResourceUserService extends KalturaBaseService
+{
+    public function initService($serviceId, $serviceName, $actionName)
+    {
+        parent::initService($serviceId, $serviceName, $actionName);
+        $this->applyPartnerFilterForClass('ResourceUser');
+    }
+
+    /**
+     * Allows you to add a new KalturaScheduleResource object
+     *
+     * @action add
+     * @param KalturaResourceUser $resourceUser
+     * @return KalturaResourceUser
+     */
+    public function addAction(KalturaResourceUser $resourceUser)
+    {
+	    // save in database
+	    $dbScheduleResource = $resourceUser->toInsertableObject();
+	    $dbScheduleResource->save();
+
+	    // return the saved object
+	    $scheduleResource = new KalturaResourceUser();
+	    $scheduleResource->fromObject($dbScheduleResource, $this->getResponseProfile());
+	    return $scheduleResource;
+    }
+
+
+    /**
+     * Allows you to delete an existing KalturaScheduleResource object
+     *
+     * @action delete
+     * @param string $resourceTag
+     * @param string $userId
+     * @return KalturaScheduleResource
+     */
+    public function deleteAction($resourceTag, $userId)
+    {
+		$partnerId = kCurrentContext::$partner_id ? kCurrentContext::$partner_id : kCurrentContext::$ks_partner_id;
+		$kuser = kuserPeer::getKuserByPartnerAndUid($partnerId, $userId);
+
+	    if (!$kuser)
+	    {
+		    if (kCurrentContext::$master_partner_id != Partner::BATCH_PARTNER_ID)
```

Comment:
are you going to work with the batch partner for this?
also need to add {}

Reply:
@inbal-ben-david it doesn't sound impossible that we might need a batch deletion process for these objects sometime in the future.